src/pip/_internal/build_env.py

```diff
@@ -204,16 +209,32 @@ class NoOpBuildEnvironment(BuildEnvironment):
     """
 
     def __init__(self):
+        # type: () -> None
         pass
 
     def __enter__(self):
+        # type: () -> None
         pass
 
-    def __exit__(self, exc_type, exc_val, exc_tb):
+    def __exit__(
+        self,
+        exc_type,  # type: Optional[Type[BaseException]]
+        exc_val,  # type: Optional[BaseException]
+        exc_tb  # type: Optional[TracebackType]
+    ):
+        # type: (...) -> bool
         pass
 
     def cleanup(self):
+        # type: () -> None
         pass
 
-    def install_requirements(self, finder, requirements, prefix, message):
+    def install_requirements(
+        self,
+        finder,  # type: PackageFinder
+        requirements,  # type: Iterable[str]
+        prefix_as_string,  # type: str
+        message  # type: Optional[str]
```

Comment:
Not sure, but doesn't keep it as `Optional` means, there should be some default value? Please correct if I am wrong.

Reply:
So I actually mirrored the types defined the types in the superclass at

https://github.com/pypa/pip/blob/76b865155ee646927e838c377b46643b51ccd2fd/src/pip/_internal/build_env.py#L157-L163

Also IIUC, keeping it [Optional](https://docs.python.org/3/library/typing.html#typing.Optional) means that `message` might be set to `None` in some function calls to `install_requirements`

Reply:
Ah! thanks. :)